devicetypes/smartthings/zigbee-multi-switch-power.src/zigbee-multi-switch-power.groovy

```diff
@@ -0,0 +1,183 @@
+		capability "Actuator"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Switch"
+		capability "Power Meter"
+
+		command "childOn", ["string"]
+		command "childOff", ["string"]
+
+		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0B04", outClusters: "0019", manufacturer: "Aurora", model: "DoubleSocket50AU", deviceJoinName: "Aurora Smart Double Socket 1"
+	}
+	// simulator metadata
+	simulator {
+		// status messages
+		status "on": "on/off: 1"
+		status "off": "on/off: 0"
+
+		// reply messages
+		reply "zcl on-off on": "on/off: 1"
+		reply "zcl on-off off": "on/off: 0"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label: '${name}', action: "switch.off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "off", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "turningOn", label: '${name}', action: "switch.off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("power", key: "SECONDARY_CONTROL") {
+				attributeState "power", label: '${currentValue} W'
+			}
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "switch"
+		details(["switch", "refresh", "power"])
+	}
+}
+
+def installed() {
+	createChildDevices()
+	updateDataValue("onOff", "catchall")
+}
+
+def updated() {
+	log.debug "updated()"
+	updateDataValue("onOff", "catchall")
+}
+
+def parse(String description) {
+	Map eventMap = zigbee.getEvent(description)
+	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+
+	if (!eventMap && eventDescMap) {
+		eventMap = [:]
+		if (eventDescMap?.clusterId == zigbee.ONOFF_CLUSTER) {
+			eventMap[name] = "switch"
+			eventMap[value] = eventDescMap?.value
+		}
+	}
+
+	if (eventMap) {
+		if (eventDescMap?.sourceEndpoint == "01" || eventDescMap?.endpoint == "01") {
+			sendEvent(eventMap)
+		} else {
+			def childDevice = childDevices.find {
+				it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.sourceEndpoint}" || it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.endpoint}"
+			}
+			if (childDevice) {
+				childDevice.sendEvent(eventMap)
+			} else {
+				log.debug "Child device: $device.deviceNetworkId:${eventDescMap.sourceEndpoint} was not found"
+			}
+		}
+	}
+}
+
+private void createChildDevices() {
+	def i = 2
```

Comment:
please write this DTH in a way that it can handle multiple devices with different numbers of child devices

Reply:
Sure. 
Most of your comments in this PR regard code that is already working in zigbee-multi-switch DTH.
Did You also mean that we should do a refactoring of zigbee-multi-switch DTH ?

Reply:
No. If it's not broken we shouldn't fix it, but if we have an opportunity to start from scratch we should do it as best we can.

Reply:
ok, some refactoring was done. 
Now it's all based on the method that returns number of child devices depending of device model name.